src/policy_engine/opa/mod.rs

```diff
@@ -23,36 +25,46 @@ pub struct GoString {
 
 #[derive(Debug)]
 pub struct OPA {
-    policy_file_path: PathBuf,
+    policy_dir_path: PathBuf,
 }
 
 impl OPA {
     pub fn new(work_dir: PathBuf) -> Result<Self> {
-        let mut policy_file_path = work_dir;
+        let mut policy_dir_path = work_dir;
 
-        policy_file_path.push("opa");
-        if !policy_file_path.as_path().exists() {
-            fs::create_dir_all(&policy_file_path)
+        policy_dir_path.push("opa");
+        if !policy_dir_path.as_path().exists() {
+            fs::create_dir_all(&policy_dir_path)
                 .map_err(|e| anyhow!("Create policy dir failed: {:?}", e))?;
         }
 
-        policy_file_path.push("policy.rego");
-        if !policy_file_path.as_path().exists() {
+        let mut default_policy_path = PathBuf::from(&policy_dir_path.to_str().unwrap());
+        default_policy_path.push("default.rego");
+        if !default_policy_path.as_path().exists() {
             let policy = std::include_str!("default_policy.rego").to_string();
-            fs::write(&policy_file_path, policy)?;
+            fs::write(&default_policy_path, policy)?;
```

Comment:
Would you like to add a new commit to change the `fs::write` to async (tokio?)

Reply:
I don't think we should change the `new()` method of OPA to aysnc...